### ISSUE TITLE
"Unsupported Content Type" for local files if not .gmi

### ISSUE THREAD

**Issue by bortzmeyer** [skyjake/lagrange#267]
If a local file (URL scheme `file:// `) does not end in `.gmi` Lagrange cannot open it and says:

```
Unsupported Content Type
The received content cannot be viewed with this application.

application/octet-stream

You can save it as a file to your Downloads folder, though. Press Ctrl+S or select "Save to Downloads" from the menu.`
```

I do not find explanations about how to define a default type (`text/gemini` for instance) or how to add new file extensions.

IMHO, when the file is a text file (as detected by libfile), Lagrange could at least open the raw text.

**Comment by skyjake** [comment]
The type of local files is currently determined solely based on the file extension, and the known extensions are hardcoded.

I suppose file type identification could be handled via external libraries, but I would really prefer not to add a dependency for this. Detecting valid UTF-8 text content is doable certainly, and perhaps some simple heuristic could be added to detect gemtext as well.

**Comment by bortzmeyer** [comment]
A (related?) issue is that, when the link goes to a directory, Lagrange displays the same "Unsupported Content Type" message, it does not try to load `index.gmi` or to display a list of files.

**Comment by skyjake** [comment]
That is somewhat related, yes. However, viewing directory contents (also inside ZIP archives) has already been implemented for v1.4 and will be available later this week.

**Comment by skyjake** [comment]
Implemented in v1.6.4: local file contents are checked for UTF-8 text regardless of file extension.
